### ISSUE TITLE
Code optimization for the offline shipping methods.

### ISSUE THREAD

**Issue by atwixfirster** [magento/magento2#23143]
### Summary (*)

`Flat Rate`, `Free Shipping`,  and `Store Pickup` shipping methods use protected $_code variable to provide information about carrier's code.

For example, `Flat Rate` shipping method:
```
class Flatrate extends AbstractCarrier implements CarrierInterface
{
    /**
     * @var string
     */
    protected $_code = 'flatrate';

...
```

In the `getAllowedMethods()` function for this shipping method we have:
```
 * @return array
     */
    public function getAllowedMethods()
    {
        return ['flatrate' => $this->getConfigData('name')];
    }
```

So, carrier's code is duplicated.

### Proposed solution

Use `$this->_code` in the `getAllowedMethods()` function of `Flat Rate`, `Free Shipping`,  and `Store Pickup`.

For example, a code optimization for `Flat Rate` will look like:
```
/**
     * @return array
     */
    public function getAllowedMethods()
    {
        return [$this->_code => $this->getConfigData('name')];
    }
```

**Comment by orlangur** [comment]
@atwixfirster such improvements may be proposed directly as pull requests without even reporting an issue ;) Issues are mostly useful when we deal with a bug and need to verify it against all release lines.

**Comment by atwixfirster** [comment]
> @atwixfirster such improvements may be proposed directly as pull requests without even reporting an issue ;) Issues are mostly useful when we deal with a bug and need to verify it against all release lines.

Thanks for the flow explanation, @orlangur ! I will use it in my future PRs. 

👍

**Comment by magento-engcom-team** [comment]
Hi @atwixfirster. Thank you for your report. 
The issue has been fixed in magento/magento2#23144 by @atwixfirster in 2.3-develop branch
Related commit(s):
 - [85bd3ec21912f464117f7a696fb60d4e8cdd06fe](https://github.com/magento/magento2/commit/85bd3ec21912f464117f7a696fb60d4e8cdd06fe)
 - [dda454d8f18d10e4aa62facbae65041fabb4d25a](https://github.com/magento/magento2/commit/dda454d8f18d10e4aa62facbae65041fabb4d25a)

The fix will be available with the upcoming 2.3.4 release.
